src/components/RunButton.js

```diff
@@ -42,13 +43,17 @@ export default function RunButton(props) {
         !outPackage.profiles.length &&
         !outPackage.valueSets.length
       ) {
+        isObject = false;
         jsonOutput = 'Your FSH is invalid. Just keep swimming!';
       }
     } else {
+      isObject = false;
       jsonOutput = 'Your FSH is invalid. Just keep swimming!';
     }
 
-    props.onClick(true, jsonOutput);
+    //This double call to onClick just resets the doRunSUSHI variable so that react doesn't get confused with different incoming text
+    props.onClick(false, jsonOutput, isObject);
+    props.onClick(true, jsonOutput, isObject);
```

Comment:
I know we talked briefly about the need to have this call to `props.onClick` twice, but I was looking into why it was needed, and I can't figure out when it is needed. Do you have a specific series of steps that I can do in the app to reproduce the issue if the first call with `false` isn't there?

Reply:
Sure! If you open up the page
1) Put in good FSH:
```
Profile:        FishPatient
Parent:         Patient
Id:             fish-patient
Title:          "Fish Patient"
Description:    "A patient that is a type of fish."
``` 
2) Don't reload, just backspace out some of that good FSH to make it bad FSH:
```
Profile:        FishPatient
Parent:         Patient
Id:             fish-patient
Title:          "Fish Patient"
Description:    "A patient that is
```
For me I get a React can't find error. Something about the parsing of the JSON will get messed up unless you reset it.

Reply:
So I didn't exactly figure out why the error comes up, but I did have another idea. What if instead of the double call at the end of the function, we move it to the top. Then we could reset the text to say "Loading" which would give some indication that SUSHI is running! Something like this at the top of the function:

```
props.onClick(true, 'Loading...', false);
```